src/backend/replication/walreceiver.c

```diff
@@ -598,10 +602,10 @@ WalReceiverMain(void)
 			 * being archived later.
 			 */
 			XLogFileName(xlogfname, recvFileTLI, recvSegNo, wal_segment_size);
-			if (XLogArchiveMode != ARCHIVE_MODE_ALWAYS)
-				XLogArchiveForceDone(xlogfname);
-			else
+			if (XLogArchiveMode == ARCHIVE_MODE_ALWAYS || wal_sender_archiving_status_interval > 0)
```

Comment:
We need to check for something (probably `archive_command` GUC) to prevent .ready files being created when archiving is not involved to prevent possible regressions for users that are not utilizing WAL archiving.

```suggestion
			if (XLogArchiveMode == ARCHIVE_MODE_ALWAYS ||
				(wal_sender_archiving_status_interval > 0 && XLogArchiveCommandSet()))
```

Reply:
There is a existing macro `XLogArchivingActive()` which checks if the `archive_mode > off`. 
I think we can use this to check if archiving is enabled. Have updated the comment as per this.

Reply:
Sure, `XLogArchivingActive()` seems better than `XLogArchiveCommandSet()`.